samples/excel/89-preview-apis/events-worksheet-activated.yaml

```diff
@@ -0,0 +1,156 @@
+order: 7
+id: excel-events-worksheet-activated
+name: Events - Worksheet activated
+description: Add an event handler on the worksheet activated and deactivated events
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |+
+        $("#setup").click(() => tryCatch(setup));
+        $("#activate-event-handler").click(() => tryCatch(onActivateEventHandler));
+        $("#deactivate-event-handler").click(() => tryCatch(onDeactivateEventHandler));
+        $("#delete-worksheet").click(() => tryCatch(deleteWorksheet));
+
+        async function onActivateEventHandler() {
+            await Excel.run(async (context) => {
+                let sheet = context.workbook.worksheets;
+                sheet.onActivated.add(onActivate);
+
+                await context.sync();
+
+                OfficeHelpers.UI.notify("Registered worksheet on activated event handler.");   
+            });
+        }
+
+        function onActivate(args) {
```

Comment:
You MUST have the keyword `async `in front of 
`function onActivate(args)` 
If you don't, the line 
`sheet.onActivated.add(onActivate);` 
which is up above gets a red squiggly underneath the `onActivate `and the **onDeactivate** event fires twice.